File: P1/E1/FactoriaMontana.java
package E1;

public class FactoriaMontana implements FactoriaCarreraYBicicleta {

    @Override
    public Bicicleta CrearBicicleta(){
        return new BicicletaMontana();
    }

    @Override
    public Carrera CrearCarrera(){
        return new CarreraMontana();
    }
}

File: P1/E2/FactoriaMontana.py
from BicicletaMontana import BicicletaMontana
from CarreraMontana import CarreraMontana
from FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta

class FactoriaMontana(FactoriaCarreraYBicicleta):
    def crearBicicleta(self, id: int):
        return BicicletaMontana(id)

    def crearCarrera(self):
        return CarreraMontana()
